src/call.cpp

```diff
@@ -923,33 +923,72 @@ char * call::get_last_header(const char * name)
     if (name[len - 1] == ':') {
         return get_header(last_recv_msg, name, false);
     } else {
+
         char with_colon[MAX_HEADER_LEN];
-        sprintf(with_colon, "%s:", name);
-        return get_header(last_recv_msg, with_colon, false);
+        const char *sep = strrchr(name, ':');
+
+        if (!sep){
+            sprintf(with_colon, "%s:", name);
+            return get_header(last_recv_msg, with_colon, false);
+        } else {
+            if (!strcmp(sep, ":value")) {
+                snprintf(with_colon, len - 4, "%s", name);
+                return get_header(last_recv_msg, with_colon, true);
+            }
+            if (!strcmp(sep, ":uri")) {
+                snprintf(with_colon, len - 2, "%s", name);
+                char * last_header_uri = get_last_request_uri(with_colon);
+                char * ret;
+                if (!(ret = (char *)malloc(strlen(last_header_uri + 1)))){
+                    ERROR("call::get_last_header: Cannot allocate uri !\n");
+                }
+                sprintf(ret, "%s", last_header_uri);
+                free(last_header_uri);
+                return ret;
+            }
+            if (!strcmp(sep, ":param.tag")) {
+                snprintf(with_colon, len - 8, "%s", name);
+                char * last_tag = get_last_tag(with_colon);
+                char * ret;
+                if (!(ret = (char *)malloc(strlen(last_tag + 1)))){
+                    ERROR("call::get_last_header: Cannot allocate param.tag !\n");
+                }
+                sprintf(ret, "%s", last_tag);
+                free(last_tag);
+                return ret;
+            }
+            ERROR("Token %s not valid in %s", sep, name);
+        }
```

Comment:
This function is too big. The code in the if(strcmp..){...} blocks should be delegated to helper methods and duplicate code should be avoided. 

For example:
```
        if (!sep) { /* [last_Header] */
            sprintf(with_colon, "%s:", name);
            return get_header(last_recv_msg, with_colon, false);
        } 

        /* [last_Header:...] */
        sprintf(with_colon, "%.*s", (sep - name + 1), name);
        if (!strcmp(sep, ":")) {
            return get_header(last_recv_msg, with_colon, false);
        }

        char *value = get_header(last_recv_msg, with_colon, true);
        if (!strcmp(sep, ":value")) {
            ; /* done */
        } else if (!strcmp(sep, ":uri")) {
            extract_header_uri(value); /* reuse mem in value (memmove?) */
        } else if (....) {
            ...
        } else {
            ERROR("Token %s not valid in %s", sep, name);
        }

        return value;
```